bokeh/charts/stats.py

```diff
@@ -387,6 +387,59 @@ def sort(self, ascending=True):
                                     reverse=~ascending))
 
 
+class Histogram(Bins):
+    """Bins and aggregates dimensions for plotting.
+
+    Takes the inputs and produces a list of bins that can be iterated over and
+    inspected for their metadata. The bins provide easy access to consistent labeling,
+    bounds, and values.
+    """
+
+    density = Bool(False, help="""
+    Whether to normalize the histogram. (default: True)
+
+    If True, the result is the value of the probability *density* function
+    at the bin, normalized such that the *integral* over the range is 1. If
+    False, the result will contain the number of samples in each bin.
+
+    For more info check ``numpy.histogram`` function documentation.
+
+    """)
+
+
+    def calculate(self):
+        bin_str = '_bin'
+        self.bin_column = self.column + bin_str
+
+        data = self.bin_stat.get_data()
+        bins = self.bin_stat.bin_count
+
+        # Choose bin bounds when data range is ill-defined; pd.cut()
+        # does not handle this well for values that are <= 0
+        if data.size < 2:
+            raise ValueError('Histogram data must have at least two elements.')
+        if data.ndim == 1 and data.std() == 0:
+            margin = 0.01 * abs(float(data[0])) or 0.01
+            bins = np.linspace(data[0] - margin, data[0] + margin, bins+1)
```

Comment:
Done. I verified with the newly added test and with a simple example plot, that the above trickery can be removed, because `np.histogram` deals with it ok. It can even handle data with just one value. So I'd propose to only leave a check for `if data.size ==0: raise ...`.